### ISSUE TITLE
Prune fails with pack: negative content length

### ISSUE THREAD

**Issue by baernm** [restic/restic#1379]
## Output of `restic version`
```
restic 0.7.3
compiled with go1.9 on linux/amd64
```

## How did you run restic exactly?

`restic -r rest:https://path forget --keep-daily 7 --keep-weekly 5 --keep-monthly 12 --prune`

```
snapshots for (host [], paths []):

keep 10 snapshots:
ID        Date                 Host        Tags        Directory
----------------------------------------------------------------------
835c6288  2017-10-20 01:24:00     1           

snapshots for (host [], paths []):

keep 10 snapshots:
ID        Date                 Host        Tags        Directory
----------------------------------------------------------------------
608df4b2  2017-10-23 01:16:53        2     


remove 3 snapshots:
ID        Date                 Host        Tags        Directory
----------------------------------------------------------------------
8d98032b  2017-10-16 01:16:59     2         

3 snapshots have been removed, running prune
counting files in repo
building new index for repo
pack file cannot be listed 8982bf97: negative content length
[0:30] 100.00%  6056 / 6056 packs
repository contains 6055 packs (50555 blobs) with 27.677 GiB bytes
processed 50555 blobs: 0 duplicate blobs, 0B duplicate
load all snapshots
find data that is still in use for 20 snapshots
[0:16] 100.00%  20 / 20 snapshots
found 49487 of 50555 data blobs still in use, removing 1068 blobs
will remove 0 invalid files
will delete 97 packs and rewrite 168 packs, this frees 791.116 MiB
[2:10] 100.00%  168 / 168 packs rewritten
counting files in repo
pack file cannot be listed 8982bf97: negative content length
[0:33] 100.00%  5876 / 5876 packs
finding old index files
saved new index as 9388c61d
remove 1 old index files
[0:00] 100.00%  265 / 265 packs deleted
done
```

`restic -r rest:https://path rebuild-index`

```
counting files in repo
pack file cannot be listed 8982bf97: negative content length
[0:18] 100.00%  5876 / 5876 packs
finding old index files
saved new index as 8e34e092
remove 1 old index files
```

`restic -r rest:https://path check`

```
Create exclusive lock for repository
Load indexes
Check all packs
pack 8982bf979302cf2701e737f9180bfd45bf37d19ba8566287459c069634be8db4: not referenced in any index
Check snapshots, trees and blobs
Fatal: repository contains errors
```

## What backend/server/service did you use?

rest-server

## Expected behavior

no errors :) Or some more information about a possible solution for this issue

## Actual behavior

the pack file is actually a null byte size file in the linux filesystem

## Steps to reproduce the behavior

Just execute the commands above, I actually cannot explain what caused the file to become corrupt

**Comment by fd0** [comment]
Thanks for the report, this seems to be a bug (I'm not sure if in the REST backend or the REST server), we'll figure it out.

As a workaround you can just delete the empty file.

**Comment by baernm** [comment]
Oh ok, thanks very much!

**Comment by fd0** [comment]
Ah, I think this is resolved, can you please test with restic 0.8.3 or the master branch, and the newest `rest-server`? If it doesn't work, please add a comment so we can reopen the issue. Thanks!
